Kernel/Storage/StorageManagement.cpp

```diff
@@ -241,8 +241,15 @@ MajorNumber StorageManagement::storage_type_major_number()
 }
 MinorNumber StorageManagement::generate_storage_minor_number()
 {
-    auto minor_number = s_device_minor_number.load();
-    s_device_minor_number++;
+    auto minor_number = s_storage_device_minor_number.load();
+    s_storage_device_minor_number++;
```

Comment:
Even though this code is probably only called in one place on one kernel thread currently, it runs the risk of assigning the same minor number to two storage devices during parallel execution (e.g. device discovery at OS runtime). `fetch_add(1)` will definitely be safe for all circumstances.